### PR TITLE
fix bats tests for hc-base

### PR DESCRIPTION
**Pull request by kschoche**
bats test was failing due to indexing on some tests because we've modified the clusterrole for connect-inject, this resolves it

### CODE REVIEW

**Comment by ishustava** [comment]
@kschoche looks like these tests are failing in CI. I'll wait for them to be fixed before reviewing.

**Comment by kschoche** [comment]
> @kschoche looks like these tests are failing in CI. I'll wait for them to be fixed before reviewing.
good catch it was on the wrong base branch!

File: test/unit/connect-inject-clusterrole.bats [hunk 1]
```diff
@@ -59,7 +59,7 @@ load _helpers
       --set 'global.enablePodSecurityPolicies=false' \
       . | tee /dev/stderr |
       yq -r '.rules | length' | tee /dev/stderr)
-  [ "${actual}" = "1" ]
+  [ "${actual}" = "2" ]
```
Comment (ishustava) [inline test/unit/connect-inject-clusterrole.bats:62]: could we use map/select for these tests? I think it'll both make it more resilient against future changes and more readable.

Comment (kschoche) [reply test/unit/connect-inject-clusterrole.bats:62]: sure thing, great idea!

**Pull request closed without merging**